include/correction/QnTools/CorrectionManager.hpp

```diff
@@ -130,10 +130,23 @@ class CorrectionManager {
     detectors_.AddDetector(name, type, phi, weight, radial_offset, axes, harmonics, norm);
   }
 
-  template<std::size_t N, typename FUNCTION>
+  template<std::size_t N, typename Function>
   void AddCutOnDetector(const std::string &detector_name,
                         const char *const (&variable_names)[N],
-                        FUNCTION cut_function,
+                        Function cut_function,
+                        const std::string &cut_description) {
+    std::array<std::string, N> variable_names_arr;
+    std::copy(std::begin(variable_names), std::end(variable_names), std::begin(variable_names_arr));
+    bool is_channel_wise = variable_manager_.FindVariable(variable_names_arr[0]).size() > 1;
+    detectors_.AddCut(detector_name,
+                      CallBacks::MakeCut(variable_names_arr, cut_function, cut_description),
```

Comment:
create cut directly instead of the callback

Reply:
One good thing with callbacks is that we can define cut before we initialized any variable, because as far as I understand this callback is triggered during initialization.

Reply:
I don't know if this feature is needed

Reply:
Yes that is correct. If there is a need for this use case we should of course keep it. I initially thought I could use it in the alice workflow. however this didn't work out after all. In their case an object which can be saved to a root file is needed, which is not achieved by the lambdas. One would need to save the cut formula in another way.